src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -184,97 +181,108 @@ object LinearMixedRegression {
       vds2
   }
 
-  def computeNEigs(S: DenseVector[Double], nEigs: Int, optDroppedVarianceFraction: Option[Double]): Int = {
-    val trace = S.toArray.sum
-    var i = S.length - 1
+  def computeNEigsDVF(S: DenseVector[Double], droppedVarianceFraction: Double): Int = {
+    require(0 <= droppedVarianceFraction && droppedVarianceFraction < 1)
+    val trace = sum(S)
+
+    var i = 0
     var runningSum = 0.0
-    if (optDroppedVarianceFraction.isEmpty) return nEigs
-    val target = optDroppedVarianceFraction.get * trace
-    while (i > nEigs || runningSum <= target) {
-      runningSum += S(-i)
-      i -= 1
+    val target = droppedVarianceFraction * trace
```

Comment:
It's true that I should check index on the loop. What to do though if it gets all the way through the loop and gets to the S.length thing? That's zero eigenvectors, yes? Should I just throw an error?